scipy/special/tests/test_mpmath.py

```diff
@@ -110,6 +110,13 @@ def test_hyp2f1_strange_points():
     pts = [
         (2, -1, -1, 0.7),  # expected: 2.4
         (2, -2, -2, 0.7),  # expected: 3.87
+        (1000, -2, -2, 0.7),
+        (2, -2, -2, 100),
+        (-5, -2, -2, 100),
+        (-100, -2, -2, 0.7),
+        (-100, -2, -2, -0.5),
+        (-100, -2, -2, -50),
+        (105, -2, -2, -50),
```

Comment:
Also extreme stuff like non-integers,  large numbers `1e30`, larger `|b|=|c|` etc is useful to check --- it's the corner cases that are hard. You can probably use itertools.product etc. to generate combinations to avoid making long lists.